components/json/src/main/java/datadog/json/JsonWriter.java

```diff
@@ -0,0 +1,335 @@
+  public JsonWriter value(int value) {
+    this.structure.checkValueAllowed();
+    injectCommaIfNeeded();
+    writeStringRaw(Integer.toString(value));
+    endsValue();
+    return this;
+  }
+
+  /**
+   * Writes a long as a number value.
+   *
+   * @param value The long to write.
+   * @return This writer.
+   */
+  public JsonWriter value(long value) {
+    this.structure.checkValueAllowed();
+    injectCommaIfNeeded();
+    writeStringRaw(Long.toString(value));
+    endsValue();
+    return this;
+  }
+
+  /**
+   * Writes a float as a number value.
+   *
+   * @param value The float to write.
+   * @return This writer.
+   */
+  public JsonWriter value(float value) {
+    this.structure.checkValueAllowed();
+    injectCommaIfNeeded();
+    writeStringRaw(Float.toString(value)); // TODO Handle Nan
+    endsValue();
+    return this;
+  }
+
+  /**
+   * Writes a double as a number value.
+   *
+   * @param value The value to write.
+   * @return This writer.
+   */
+  public JsonWriter value(double value) {
+    this.structure.checkValueAllowed();
+    injectCommaIfNeeded();
+    writeStringRaw(Double.toString(value)); // TODO Handle Nan
+    endsValue();
+    return this;
+  }
+
+  /**
+   * Starts a JSON array.
+   *
+   * @return This writer.
+   */
+  public JsonWriter beginArray() {
+    this.structure.beginArray();
+    injectCommaIfNeeded();
+    write('[');
+    return this;
+  }
+
+  /**
+   * Ends the current JSON array.
+   *
+   * @return This writer.
+   */
+  public JsonWriter endArray() {
+    this.structure.endArray();
+    endsValue();
+    write(']');
+    return this;
+  }
+
+  /**
+   * Gets the JSON String as a UTF-8 byte array.
+   *
+   * @return The JSON String as a UTF-8 byte array.
+   */
+  public byte[] toByteArray() {
+    flush();
+    return this.outputStream.toByteArray();
+  }
+
+  @Override
+  public String toString() {
+    return new String(toByteArray(), UTF_8);
+  }
+
+  @Override
+  public void flush() {
+    try {
+      this.writer.flush();
+    } catch (IOException ignored) {
+    }
+  }
+
+  @Override
+  public void close() {
+    try {
+      this.outputStream.close();
+      this.writer.close();
+    } catch (IOException ignored) {
+    }
+  }
+
+  private void injectCommaIfNeeded() {
+    if (this.requireComma) {
+      write(',');
+    }
+    this.requireComma = false;
+  }
+
+  private void endsValue() {
+    this.requireComma = true;
+  }
+
+  private void write(char ch) {
+    try {
+      this.writer.write(ch);
+    } catch (IOException e) {
+      // ignore
+    }
+  }
+
+  private void writeStringLiteral(String str) {
+    try {
+      this.writer.write('"');
+
+      for (int i = 0; i < str.length(); ++i) {
+        char c = str.charAt(i);
+        // Escape any char outside ASCII to their Unicode equivalent
+        if (c > 127) {
+          this.writer.write('\\');
+          this.writer.write('u');
+          String hexCharacter = Integer.toHexString(c).toUpperCase(Locale.ENGLISH);
```

Comment:
```suggestion
          String hexCharacter = Integer.toHexString(c).toUpperCase(Locale.ROOT);
```
`Locale.ROOT` is used more in the codebase

Reply:
Agreed. This originally comes from `String.hex(char ch)` (which is no more used by the may, so I will remove it too).